### PR TITLE
Fixed BM Flashes of 404 screen on Admin panel 

### PR DESCRIPTION
**Pull request by dmitrii-zolotukhin**
- [x] Remove false appearance of 404 page when going to "Add a room"

fixes for sparkletown/internal-sparkle-issues#1123

### CODE REVIEW

**Review by goran-peoski-work — CHANGES_REQUESTED** [review]
Shouldn't have to use isLoading state in this component. It could probably use a hook like `useAsyncFn` which provides it. Example:

```ts
 const [{ loading: isLoading }, fetchVenueFromAPI ] = useAsyncFn(async () => {} );
```

File: src/pages/Admin/Venue/Rooms/RoomsForm.tsx [hunk 1]
```diff
@@ -52,30 +54,31 @@ export const RoomsForm: React.FC = () => {
...
-      const template = ALL_VENUE_TEMPLATES.find(
-        (template) => data.template === template.template
-      );
+    if (!template || !HAS_ROOMS_TEMPLATES.includes(template.template)) {
+      history.replace("/admin");
+    }
+    setVenue(data as PartyMapVenue);
+  }, [firestore, history, venueId]);
+
+  useEffect(() => {
```
Comment (denisdimitrov) [inline src/pages/Admin/Venue/Rooms/RoomsForm.tsx:77]: You can use `useAsync` and get rid of this `useEffect`

**Comment by dmitrii-zolotukhin** [comment]
> Shouldn't have to use isLoading state in this component. It could probably use a hook like `useAsyncFn` which provides it. Example:
> 
> ```ts
>  const [{ loading: isLoading }, fetchVenueFromAPI ] = useAsyncFn(async () => {} );
> ```

fixed in 7d11be598f30394d024bebabd3dcf6160fe7d912

File: src/pages/Admin/Venue/Rooms/RoomsForm.tsx (hunk 1 above)
Comment (dmitrii-zolotukhin) [reply src/pages/Admin/Venue/Rooms/RoomsForm.tsx]: @denisdimitrov fixed in 8046f962fca8b50460576d6e7fad9013ae583373

**Review by goran-peoski-work — APPROVED** [review]
LGTM

**Pull request merged**